### PR TITLE
build: add prettier:check

### PR DESCRIPTION
**Pull request by aladdin-add**
refs: https://github.com/umijs/qiankun/pull/2704#discussion_r1338408941

##### Checklist

- [ ] `npm test` passes
- [ ] tests are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

- any feature?
- close https://github.com/umijs/qiankun/ISSUE_URL

### CODE REVIEW

File: packages/sandbox/src/core/sandbox/globals.ts
Comment (kuitos) [inline packages/sandbox/src/core/sandbox/globals.ts:1]: 这个文件不用提交

File: package.json [hunk 1]
```diff
@@ -14,9 +14,10 @@
     "release:alpha": "pnpm run build && changeset publish && changeset pre exit",
     "lint": "eslint packages/",
     "prettier": "prettier --write .",
+    "prettier:check": "prettier -c .",
```
Comment (kuitos) [inline package.json:17]: examples 目录过滤了么？

Comment (aladdin-add) [reply package.json:17]: 好像之前已经过滤了：https://github.com/umijs/qiankun/blob/6cf4d892836b869ebc1f940200f7458090a7b0f6/.prettierignore#L1-L9

File: packages/sandbox/src/core/sandbox/globals.ts
Comment (aladdin-add) [reply packages/sandbox/src/core/sandbox/globals.ts:1]: 加到ignore里面了：https://github.com/umijs/qiankun/pull/2707/commits/e9fcd29b6a371e6739acb60c55bd79b03022aa56

File: .prettierrc [hunk 2]
```diff
@@ -1,5 +1,4 @@
 {
-  "editorconfig": true,
```
Comment (kuitos) [inline .prettierrc:2]: 这个为啥删了？

File: .github/workflows/changeset-prerelease.yml [hunk 3]
```diff
@@ -19,7 +19,7 @@ jobs:
         uses: actions/setup-node@v3
         with:
           node-version: 18
-          cache: "pnpm"
+          cache: 'pnpm'
```
Comment (kuitos) [inline .github/workflows/changeset-prerelease.yml:22]: yaml 里还是优先双引号吧，prettier 里要单独加个分类

File: .prettierrc (hunk 2 above)
Comment (aladdin-add) [reply .prettierrc:2]: https://github.com/prettier/prettier/issues/6176#issuecomment-498808239

Comment (kuitos) [reply .prettierrc:2]: 裂开，看来是不打算修了 https://github.com/prettier/prettier/issues/15401

File: .github/workflows/changeset-prerelease.yml (hunk 3 above)
Comment (aladdin-add) [reply .github/workflows/changeset-prerelease.yml]: https://github.com/umijs/qiankun/pull/2707/commits/d27584e12c3e29a93c2d02442ec2159283657203

**Comment by aladdin-add** [comment]
image

vercel好像每次部署都会失败呢，这个可以单独看下。

**Review by kuitos — APPROVED** [review]
LGTM

File: .prettierrc (hunk 2 above)
Comment (fisker) [reply .prettierrc:2]: 打算，且更进一步，把所有验证都删了 https://github.com/prettier/prettier/issues/15443 😄

Comment (aladdin-add) [reply .prettierrc:2]: 还想提个PR修复来着，不过看着好像不是prettier报出的，而是某个依赖，没太看懂。😅

**Pull request merged**